bindep.txt

```diff
@@ -1,6 +1,10 @@
-libxml2-devel [platform:rpm]
+openssl-devel [platform:rpm compile]
+gcc [platform:rpm compile]
+libcurl-devel [platform:rpm compile]
+libxml2-devel [platform:rpm compile]
```

Comment:
Why are those libraries for? According to the RPM spec, which is primary distribution channel for oVirt/RHV we don't need those libraries:

https://github.com/oVirt/ovirt-ansible-collection/blob/master/ovirt-ansible-collection.spec.in#L14

Reply:
`libxml2-devel` was already here, I have just designated it in this PR as a compile-time dependency.

For the others, to be completely verbose..

The error without `libcurl-devel`:

```
Collecting pycurl>=7.19.0
  Downloading pycurl-7.43.0.6.tar.gz (222 kB)
    ERROR: Command errored out with exit status 1:
     command: /tmp/venv/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-aq4afh74/pycurl_bca620fded8149899be0c6b19cbe126a/setup.py'"'"'; __file__='"'"'/tmp/pip-install-aq4afh74/pycurl_bca620fded8149899be0c6b19cbe126a/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-h7899v9d
         cwd: /tmp/pip-install-aq4afh74/pycurl_bca620fded8149899be0c6b19cbe126a/
    Complete output (22 lines):
    Traceback (most recent call last):
      File "/tmp/pip-install-aq4afh74/pycurl_bca620fded8149899be0c6b19cbe126a/setup.py", line 237, in configure_unix
        stdout=subprocess.PIPE, stderr=subprocess.PIPE)
      File "/usr/lib64/python3.6/subprocess.py", line 729, in __init__
        restore_signals, start_new_session)
      File "/usr/lib64/python3.6/subprocess.py", line 1364, in _execute_child
        raise child_exception_type(errno_num, err_msg, err_filename)
    FileNotFoundError: [Errno 2] No such file or directory: 'curl-config': 'curl-config'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-aq4afh74/pycurl_bca620fded8149899be0c6b19cbe126a/setup.py", line 988, in <module>
        ext = get_extension(sys.argv, split_extension_source=split_extension_source)
      File "/tmp/pip-install-aq4afh74/pycurl_bca620fded8149899be0c6b19cbe126a/setup.py", line 649, in get_extension
        ext_config = ExtensionConfiguration(argv)
      File "/tmp/pip-install-aq4afh74/pycurl_bca620fded8149899be0c6b19cbe126a/setup.py", line 101, in __init__
        self.configure()
      File "/tmp/pip-install-aq4afh74/pycurl_bca620fded8149899be0c6b19cbe126a/setup.py", line 241, in configure_unix
        raise ConfigurationError(msg)
    __main__.ConfigurationError: Could not run curl-config: [Errno 2] No such file or directory: 'curl-config': 'curl-config'
```


The error without gcc:

```
Collecting ovirt-engine-sdk-python>=4.4.0
  Downloading ovirt-engine-sdk-python-4.4.9.tar.gz (331 kB)
Collecting pycurl>=7.19.0
  Downloading pycurl-7.43.0.6.tar.gz (222 kB)
Collecting six
  Downloading six-1.15.0-py2.py3-none-any.whl (10 kB)
Building wheels for collected packages: ovirt-engine-sdk-python, pycurl
  Building wheel for ovirt-engine-sdk-python (setup.py): started
  Building wheel for ovirt-engine-sdk-python (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /tmp/venv/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3l28rwy0/ovirt-engine-sdk-python_98535b3b06ad4873876974102f7b0771/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3l28rwy0/ovirt-engine-sdk-python_98535b3b06ad4873876974102f7b0771/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-01anzp3o
       cwd: /tmp/pip-install-3l28rwy0/ovirt-engine-sdk-python_98535b3b06ad4873876974102f7b0771/
  Complete output (24 lines):
  4.4.9
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/__init__.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/http.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/reader.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/readers.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/service.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/services.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/types.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/version.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/writer.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  copying lib/ovirtsdk4/writers.py -> build/lib.linux-x86_64-3.6/ovirtsdk4
  running build_ext
  building 'ovirtsdk4.xml' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/ext
  gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/libxml2 -I/tmp/venv/include -I/usr/include/python3.6m -c ext/ov_xml_module.c -o build/temp.linux-x86_64-3.6/ext/ov_xml_module.o
  unable to execute 'gcc': No such file or directory
  error: command 'gcc' failed with exit status 1
```

The error without openssl-devel:

```
  Building wheel for pycurl (setup.py): started
  Building wheel for pycurl (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /tmp/venv/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-6he63lan/pycurl_453e22d1f9f5432b8d5c62c74dc7abc4/setup.py'"'"'; __file__='"'"'/tmp/pip-install-6he63lan/pycurl_453e22d1f9f5432b8d5c62c74dc7abc4/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-qqp0yvfx
       cwd: /tmp/pip-install-6he63lan/pycurl_453e22d1f9f5432b8d5c62c74dc7abc4/
  Complete output (20 lines):
  Using curl-config (libcurl 7.61.1)
  Using SSL library: OpenSSL/LibreSSL/BoringSSL
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/curl
  copying python/curl/__init__.py -> build/lib.linux-x86_64-3.6/curl
  running build_ext
  building 'pycurl' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/src
  gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DPYCURL_VERSION="7.43.0.6" -DHAVE_CURL_SSL=1 -DHAVE_CURL_OPENSSL=1 -DHAVE_CURL_SSL=1 -I/tmp/venv/include -I/usr/include/python3.6m -c src/docstrings.c -o build/temp.linux-x86_64-3.6/src/docstrings.o
  In file included from src/docstrings.c:4:
  src/pycurl.h:174:13: fatal error: openssl/ssl.h: No such file or directory
   #   include <openssl/ssl.h>
               ^~~~~~~~~~~~~~~
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pycurl
  Running setup.py clean for pycurl
Successfully built ovirt-engine-sdk-python
Failed to build pycurl
Installing collected packages: six, pycurl, ovirt-engine-sdk-python
    Running setup.py install for pycurl: started
    Running setup.py install for pycurl: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /tmp/venv/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-6he63lan/pycurl_453e22d1f9f5432b8d5c62c74dc7abc4/setup.py'"'"'; __file__='"'"'/tmp/pip-install-6he63lan/pycurl_453e22d1f9f5432b8d5c62c74dc7abc4/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-fq2h5suh/install-record.txt --single-version-externally-managed --compile --install-headers /tmp/venv/include/site/python3.6/pycurl
         cwd: /tmp/pip-install-6he63lan/pycurl_453e22d1f9f5432b8d5c62c74dc7abc4/
    Complete output (20 lines):
    Using curl-config (libcurl 7.61.1)
    Using SSL library: OpenSSL/LibreSSL/BoringSSL
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/curl
    copying python/curl/__init__.py -> build/lib.linux-x86_64-3.6/curl
    running build_ext
    building 'pycurl' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/src
    gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DPYCURL_VERSION="7.43.0.6" -DHAVE_CURL_SSL=1 -DHAVE_CURL_OPENSSL=1 -DHAVE_CURL_SSL=1 -I/tmp/venv/include -I/usr/include/python3.6m -c src/docstrings.c -o build/temp.linux-x86_64-3.6/src/docstrings.o
    In file included from src/docstrings.c:4:
    src/pycurl.h:174:13: fatal error: openssl/ssl.h: No such file or directory
     #   include <openssl/ssl.h>
```

Reply:
Maybe would be solved by installing the oVirt SDK from RPM.

Reply:
is the ovirt sdk RPM in centos today? And are you okay with that version over pulling it from pypi?  if so, then yes, removing it from requirements.txt into bindep.txt should just work.

Reply:
We have the SDK RPM in our repo oVirt repo [1] and we even prefer it from the PyPI installation. The PyPI is just a feature that is not officially supported by us. 
[1] https://plain.resources.ovirt.org/pub/ovirt-4.4/rpm/el8/

Reply:
@pabelanger is it possible for us to enable this repo and install this through the `bindep.txt`?

We have the epel repo that we can optionally use.

Reply:
@AlanCoding yup, we'd have to do that in the prepend stage of ansible-builder. Once enabled, the package becomes available to bindep.

Reply:
actually, do we have rpms for centos-8?

Reply:
but does it make sense to enable the repo in a prepend step but still put the requirement in `bindep.txt` here? That just means that it'll break any execution environment that includes the collection without the prepend steps. It would be better to add the repo in prepend steps _and_ put the requirement in the user's own `bindep.txt`, and not include anything in the collection itself.

Reply:
Right, that is true.  Bindep doesn't manage if a repo is enable or disabled. So, you can still put it here, with some sort of comment, assuming the repo avoid is accessible.

In the case of EEs, that should always be the case if awx-ee added the prepend step.